docs/software_requirements/timers.sdoc

```diff
@@ -26,12 +23,334 @@ UID: ZEP-SRS-4-2
+UID: ZEP-SRS-4-15
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer status read
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to read the status of a timer.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-16
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer status read reset
+STATEMENT: >>>
+When the timer is read via the timer status read mechanism, the status shall be reset.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-17
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer status read reset
+STATEMENT: >>>
+When the timer is read via the timer status read mechanism, the status shall be reset.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-18
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer thread synchronization
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to synchronize a thread with a defined and initialized timer.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-19
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer expire when thread synchronization is used
+STATEMENT: >>>
+When the thread synchronization mechanism is used, the thread shall be blocked until the timer expires.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-20
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer is stopped when thread synchronization is used
+STATEMENT: >>>
+When the thread synchronization mechanism is used, the thread shall be blocked until the timer is stopped.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-21
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer thread synchronization status reset
+STATEMENT: >>>
+When the thread synchronization mechanism is used, the status of the timer shall be reset.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-22
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer remaining expiration system tick time
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to get the timer remaining until expiration time in system ticks.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-23
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer remaining expiration time
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to get the timer remaining until expiration time in milliseconds.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-24
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer remaining expiration time
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to get the timer remaining until expiration time in milliseconds.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
```

Comment:
**question:** Is this a duplicate of ZEP-SRS-4-23?

Reply:
removed